src/Filesystem/FilesystemInterface.php

```diff
@@ -37,7 +37,7 @@ public function copy(string $originFile, string $targetFile, bool $override = fa
      *
      * @throws IOException On any directory creation failure
      */
-    public function mkdir($dirs, int $mode = 0777);
+    public function mkdir(iterable|string $dirs, int $mode = 0777) ;
```

Comment:
why these changes are necessary?

Reply:
The class `Symfony\Component\Filesytem\Filesystem` has been changed.
This class is the parent of `Sylius\Bundle\ThemeBundle\Filesystem\Filesystem`. And this class implements this interface.

Reply:
Indeed, but this is very strange why there is a FilesystemInterface in this package here...
While not using Symfony interface directly or extending it... I don't know this package at all, but if we change this interface, this is indeed a bc-break and we should make a new release to add support for Symfony 6.

@Roshyo What do you think?

Reply:
@macintoshplus Could you add a note on UPGRADE.md for the interface changes?

Reply:
@loic425 I have just updated the `Upgrade.md` file.

Reply:
Alternative to avoid future break the `Filesystem` implementation could inject the `filesystem` service instead of extending it: https://github.com/Sylius/SyliusThemeBundle/blob/2741cd68a8c056d56a5472445786832637c29704/src/Filesystem/Filesystem.php#L18-L24

This way the interface could stay as it is and as the Filesystem is final nobody extended from it and so no bc break would be required in this interface.